src/System.IO.Pipelines/src/System/IO/Pipelines/PipeReader.cs

```diff
@@ -62,5 +63,68 @@ public abstract class PipeReader
         /// Cancel the pending <see cref="ReadAsync"/> operation. If there is none, cancels next <see cref="ReadAsync"/> operation, without completing the <see cref="PipeWriter"/>.
         /// </summary>
         public abstract void OnWriterCompleted(Action<Exception, object> callback, object state);
+
+        /// <summary>
+        /// Asynchronously reads the bytes from the <see cref="PipeReader"/> and writes them to the specified stream, using a specified buffer size and cancellation token.
+        /// </summary>
+        /// <param name="destination">The stream to which the contents of the current stream will be copied.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A task that represents the asynchronous copy operation.</returns>
+        public virtual async Task CopyToAsync(Stream destination, CancellationToken cancellationToken = default)
```

Comment:
This should be:
```C#
public virtual Task CopyToAsync(Stream destination, CancellationToken cancellationToken = default)
{
    if (destination == null)
    {
        throw new ArgumentNullException(nameof(destination));
    }
    if (cancellationToken.IsCancellationRequested)
    {
        return Task.FromCanceled(CancellationToken);
    }
    return CopyToAsyncCore(destination, cancellationToken);

    … // current method as a local function
}
```
or something along those lines.

Reply:
Ah yes, error handling and what not 😄